server/transcribe-server.py

```diff
@@ -1,42 +1,115 @@
-from flask import Flask, request, jsonify
-import json
-
+import argparse
 import base64
-import numpy as np
-import tempfile
-import torch
-import torchaudio
+from enum import Enum
 import io
-
 import os
+from threading import Semaphore
 import threading
 
 from faster_whisper import WhisperModel
+from flask import Flask, request, jsonify
+import tempfile
+
+from server_models import (
+    ResponseFormat,
+    CreateTranscriptionRequest,
+    segments_to_response,
+)
 
 app = Flask(__name__)
-app.config['MAX_CONTENT_LENGTH'] = 3 * 1024 * 1024  # 1 Megabyte
+app.config["MAX_CONTENT_LENGTH"] = 3 * 1024 * 1024  # 1 Megabyte
 
-from threading import Semaphore
 
 MAX_CONCURRENT_TASKS = 2
 task_semaphore = Semaphore(MAX_CONCURRENT_TASKS)
 
 MAX_IN_FLIGHT_TASKS = 5000
 in_flight_semaphore = Semaphore(MAX_IN_FLIGHT_TASKS)
 
-whisper_models = [] 
+whisper_models = []
+
+
+@app.route("/v1/audio/transcriptions", methods=["POST"])
+@app.route("/audio/transcriptions", methods=["POST"])
+def transcribe_audio():
+    try:
+        request_data = CreateTranscriptionRequest(
+            **{
+                "file": request.files["file"].stream.read(),
+                "model": request.form.get("model", "whisper-1"),
+                "response_format": request.form.get("response_format", "json"),
+                "language": request.form.get("language"),
+                "prompt": request.form.get("prompt"),
+            }
+        )
+
+        if request.form.get("temperature") is not None:
+            request_data.temperature = float(request.form.get("temperature"))
+        if request.form.get("timestamp_granularities[]") is not None:
+            request_data.timestamp_granularities = request.form.getlist(
+                "timestamp_granularities[]"
+            )
+    except Exception as e:
+        return str(e), 400
```

Comment:
Nicer if you can use a named constant (example: HTTPStatus.TOO_MANY_REQUESTS for 429) in relevant places.